### PR TITLE
automatic create a github release if merged to master

### PR DESCRIPTION
**Pull request by robkooper**
This workflow will use the CHANGELOG to create a github release when code is merged into master, no action from user needed.

in this case if the release in docs/src/config.py is not increased, this will result in a failed build.

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          changelog="$(sed -e "1,/^## \[${version}\]/d" -e "/^## \[/,\$d" -e '/^$/d' CHANGELOG.md)"
+          changelog="${changelog//'%'/'%25'}"
+          changelog="${changelog//$'\n'/'%0A'}"
+          changelog="${changelog//$'\r'/'%0D'}"
+          echo "::set-output name=version::$version"
+          echo "::set-output name=changelog::$changelog"
+
+      - name: create release
+        if: github.event_name != 'pull_request' && github.repository == env.MAIN_REPO
+        uses: actions/create-release@v1
```
Comment (vnarah2) [inline .github/workflows/release.yaml:35]: I'm not sure if we should be concerned about this.. `actions/create-release` is not being maintained anymore. In the near future, should we try to move to one of the actions that are active? Some alternatives are listed here: https://github.com/actions/create-release

Comment (robkooper) [reply .github/workflows/release.yaml]: nice catch, switched to https://github.com/softprops/action-gh-release

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - master
```
Comment (ywkim312) [inline .github/workflows/release.yaml:6]: this is 'main' now

**Comment by ywkim312** [comment]
This has been implmented in https://github.com/IN-CORE/pyincore-data/pull/45

**Pull request closed without merging**